views/default/forms/admin/site/advanced/system.php

```diff
@@ -33,5 +33,81 @@
 			?>
 		</label>
 	</div>
-	<?php } ?>
+	<?php }
+	
+	// cloud storage for user data
+	$label = elgg_echo('admin:settings:user_data:data_store');
+	$user_data_store_input = elgg_view('input/dropdown', [
+		'name' => 'user_data_store',
+		'options_values' => [
+			'data_dir' => elgg_echo('admin:settings:user_data:data_dir'),
+			'aws_s3' => elgg_echo('admin:settings:user_data:aws_s3')
+		],
+		'value' => elgg_get_config('user_data_store'),
+		'id' => 'elgg-settings-user-data-store'
+	]);
+	?>
+	
+	<label>
+		<?php
+			echo $label;
+			echo $user_data_store_input;
+		?>
+	</label>
+	
+	<div class="hidden elgg-settings-user-data-store-info" id="elgg-settings-user-data-store-data-dir">
+		<p class="elgg-text-help"><?php echo elgg_echo('admin:settings:user_data:data_dir:info'); ?></p>
+	</div>
+	
+	<div class="hidden elgg-settings-user-data-store-info" id="elgg-settings-user-data-store-aws-s3">
+		<p class="elgg-text-help"><?php echo elgg_echo('admin:settings:user_data:aws_s3:info'); ?></p>
+		<?php
+			$info = elgg_get_config('user_data_store_info');
+			$s3 = elgg_extract('aws_s3', $info, []);
```

Comment:
Wouldn't mind putting the specific implementations into a plugin. This goes with #8111.